### DISCUSSION TITLE
how does TypeDeserialization work with DynamoDBStreamEvent

### DISCUSSION THREAD

**Discussion by Muthuveerappanv** [aws-powertools/powertools-lambda-python#976] [Q&A] [answered]
Trying to de-serialize dynamodb stream event in lambda, when using the below, number types get transformed to string

``` python
# event: DynamoDBStreamEvent
for record in event.records:
        if record.event_name == DynamoDBRecordEventName.REMOVE:
          output = {k: v.get_value for k, v in data.items()} 
````

so tried using type deserializer from boto3, but below didn't work as v is a object and not a dict 
``` python
from boto3.dynamodb.types import TypeDeserializer
deserializer = TypeDeserializer()
# event: DynamoDBStreamEvent
for record in event.records:
        if record.event_name == DynamoDBRecordEventName.REMOVE:
           output = {k: deserializer.deserialize(v) for k, v in data.items()}
# ERROR - AttributeError: 'AttributeValue' object has no attribute 'keys'
```

What worked - was passing the `_data` dict field to the serializer
``` python
from boto3.dynamodb.types import TypeDeserializer
deserializer = TypeDeserializer()
# event: DynamoDBStreamEvent
for record in event.records:
        if record.event_name == DynamoDBRecordEventName.REMOVE:
           output = {k: deserializer.deserialize(v._data) for k, v in data.items()}
```

Is there a way to derserialize the `AttributeValue` class using -  using `TypeDeserializer`
(When we have a huge data-set or dynamic values coming in, looking up the `AttributeValueType` for each value becomes tedious)

**Comment by michaelbrewer** [comment]
@Muthuveerappanv would it be possible to post a more complete example with an event, so i can reproduce your issue and see if we can make a fix for it or use the existing api to do what you want.

Otherwise `get_value` was added to `AttributeValue` to do the "pythonic value of the DynamoDB type" that the [TypeDeserializer](https://boto3.amazonaws.com/v1/documentation/api/latest/_modules/boto3/dynamodb/types.html) expects

**Comment by Muthuveerappanv** [comment, marked as answer]
[ddb_stream_deserialize.zip](https://github.com/awslabs/aws-lambda-powertools-python/files/8460725/ddb_stream_deserialize.zip)

**See if this helps. ☝🏻 ** 
`get_value` helps get value of individual attribute. What if I want to transform the entire record. 
As highlighted in the example, getting the _data from the object, helps get the dict and from there the deserializer can work, but i wanted to check if that was the intended way from a powertools data model perspective 

``` py
from_dict = {k: deserializer.deserialize(v) for k, v in record.dynamodb._data["OldImage"].items()}
            print("Works with Dict", from_dict)
```

**Reply by michaelbrewer** [reply]
makes sense. And it might not be obvious using `_data` is the best choice. Maybe a helper function with docs might help.
